pkg/api/validation/validation.go

```diff
@@ -1237,6 +1256,21 @@ func ValidatePersistentVolume(pv *api.PersistentVolume) field.ErrorList {
 			allErrs = append(allErrs, validateScaleIOVolumeSource(pv.Spec.ScaleIO, specPath.Child("scaleIO"))...)
 		}
 	}
+	if pv.Spec.LocalVolume != nil {
+		if numVolumes > 0 {
+			allErrs = append(allErrs, field.Forbidden(specPath.Child("localVolume"), "may not specify more than 1 volume type"))
+		} else {
+			numVolumes++
+			if !utilfeature.DefaultFeatureGate.Enabled(features.PersistentLocalVolumes) {
+				allErrs = append(allErrs, field.Forbidden(specPath.Child("localVolume"), "LocalVolumes are disabled by feature-gate"))
+			}
+			allErrs = append(allErrs, validateLocalVolumeSource(pv.Spec.LocalVolume, specPath.Child("localVolume"))...)
+			// StorageClass is required
+			if len(pv.Spec.StorageClassName) == 0 {
```

Comment:
And one more question, if it is used as a flag to change behavior of binding, what will be the trigger for unbinding?  Say we have a Pod requesting a PVC, scheduler binds it to a PV.  If we delete the Pod, I don't know if there is a path to trigger an unbind. If not, that means later Pod reusing the PVC, it will be scheduled to the specific Node. But for user, this PVC should be no difference from a new PVC created from scratch.

I think we can either do this statically, i.e. in pv controller, unbind PV/PVC if no Pod is using the PVC; or dynamically, rebind PV/PVC at schedule time. Is this discussed somewhere?